jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/LoggingMXBeanImpl.java

```diff
@@ -151,20 +209,62 @@ public String getParentLoggerName(String loggerName) {
 	 */
 	@Override
 	public void setLoggerLevel(String loggerName, String levelName) {
-		final Logger logger = LogManager.getLogManager().getLogger(loggerName);
-
+		final Object logger;
+		try {
+			logger = getLoggerFromName(loggerName);
+		} catch (Exception e) {
+			newInternalInvokeError(e);
+			return;
+		}
+		
 		if (logger != null) {
 			// The named Logger exists. Now attempt to set its level. The
 			// below attempt to parse a Level from the supplied levelName
 			// will throw an IllegalArgumentException if levelName is not
 			// a valid level name.
-			Level newLevel = Level.parse(levelName);
-			logger.setLevel(newLevel);
+			try {
+				Object newLevel = le_parse.invoke(null, levelName);
+				lo_setLevel.invoke(logger, newLevel);
+			} catch (Exception e) {
+				newInternalInvokeError(e);
+			}
+			
 		} else {
 			// Named Logger does not exist.
 			/*[MSG "K05E7", "Unable to find Logger with name {0}."]*/
 			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K05E7", loggerName)); //$NON-NLS-1$
 		}
 	}
-
+	
+	private static Object getLoggerFromName(String loggerName) throws Exception {
+		Object logManagerInstance = lm_getLogManager.invoke(null);
+		return lm_getLogger.invoke(logManagerInstance, loggerName);
+	}
+	
+	/* Handle error thrown by method invoke as internal error */
+	private static void newInternalInvokeError(Exception cause) {
```

Comment:
This should claim to return `InternalError` as it makes it easier to write the obviously correct code.